client/quantum_serverless/core/provider.py

```diff
@@ -455,19 +455,58 @@ class IBMServerlessProvider(ServerlessProvider):
     """
     A provider for connecting to the IBM serverless host.
 
-    Example:
-        >>> provider = IBMServerlessProvider(token="<TOKEN>")
+    Credentials can be saved to disk by calling the `save_account()` method::
+
+        from quantum_serverless import IBMServerlessProvider
+        IBMServerlessProvider.save_account(token=<INSERT_IBM_QUANTUM_TOKEN>)
+
+    Once the credentials are saved, you can simply instantiate the provider with no
+    constructor args, as shown below.
+
+        from quantum_serverless import IBMServerlessProvider
+        provider = IBMServerlessProvider()
+
+    Instead of saving credentials to disk, you can also set the environment variable
+    ENV_GATEWAY_PROVIDER_TOKEN and then instantiate the provider as below::
+
+        from quantum_serverless import IBMServerlessProvider
+        provider = IBMServerlessProvider()
+
+    You can also enable an account just for the current session by instantiating the
+    provider with the API token::
+
+        from quantum_serverless import IBMServerlessProvider
+        provider = IBMServerlessProvider(token=<INSERT_IBM_QUANTUM_TOKEN>)
     """
 
-    def __init__(self, token: str):
+    def __init__(self, token: Optional[str] = None, name: Optional[str] = None):
         """
         Initialize a provider with access to an IBMQ-provided remote cluster.
 
         Args:
             token: IBM quantum token
+            name: Name of the account to load
         """
+        token = token or IBMProvider(name=name).active_account().get("token")
```

Comment:
If the user passes a token, everything is normal and `name` is ignored.

If a user only passes a name, it will retrieve the token from that account.  I don't believe a user can save an account without passing a token, so it should be "guaranteed" to be there.

If a user passes neither argument, `IBMProvider` will check env variables and the default IBM Quantum json file location for credentials. An error is raised if it can't find a token in either location.

If a user passes a name of an account which doesn't exist, `IBMProvider` will raise an error.

Reply:
So much better! :) if we put it in the doc that would be it :)